registrar/apps/api/v0/views.py

```
@@ -114,3 +125,66 @@ def get_queryset(self):
+
+    Returns:
+     * 200: Returns a map of students and their enrollment status.
+     * 207: Not all students enrolled. Returns resulting enrollment status.
+     * 401: User is not authenticated
+     * 403: User lacks read access organization of specified program.
+     * 404: Program does not exist.
+     * 413: Payload too large, over 25 students supplied.
+     * 422: Invalid request, unable to enroll students.
+    """
+    authentication_classes = (JwtAuthentication, SessionAuthentication)
+    permission_classes = (IsAuthenticated,)
+    serializer_class = RequestedLearnerProgramEnrollmentSerializer
+
+    def post(self, request, *args, **kwargs):
+        """ Enroll up to 25 students in program """
+        if not self.program.managing_organization.metadata_readable:
+            raise PermissionDenied()
+
+        response = {}
+        error_status = None
+
+        if not isinstance(request.data, list):
+            raise ValidationError()
+
+        if len(request.data) > 25:
+            return Response(
+                'enrollement limit 25', HTTP_413_REQUEST_ENTITY_TOO_LARGE
+            )
+
+        for enrollee in request.data:
+            enrollee_serializer = RequestedLearnerProgramEnrollmentSerializer(
+                data=enrollee
+            )
+
+            if enrollee_serializer.is_valid():
+                enrollee = enrollee_serializer.data
+                student_key = enrollee['student_key']
+                if student_key in response:
+                    response[student_key] = 'duplicated'
+                    error_status = HTTP_207_MULTI_STATUS
+                else:
+                    response[student_key] = enrollee['status']
+            else:
+                try:
+                    if 'status' in enrollee_serializer.errors:
+                        response[enrollee['student_key']] = 'invalid-status'
+                    else:
+                        response[enrollee['student_key']] = 'internal-error'
```

I think missing `email` should raise a 422, just like missing `student_key` would. `'internal-error'` is for unexpected processing errors on our side.